### PR TITLE
[BNS] Review and update READMEs

### PR DESCRIPTION
**Pull request by ahsan-javaid**
## Description
This PR updates the `bns` package readme as mentioned in #913. There will be separate PR's for other packages readme updates to make the review process easier. 

For details refer to issue #913

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [X] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
Updated `bns` readme file

## Testing information
- Copy/Paste snippets in readme should work
- Refer to`bns` readme file

## Checklist
- [ ] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [ ] `npm run test` passes
- [ ] Changelog is updated
- [X] Tag 1 of @yknl or @zone117x for review

### CODE REVIEW

File: packages/bns/README.md [hunk 1]
```diff
@@ -12,31 +12,278 @@ state without relying on any central points of control.
...
+  createStacksPrivateKey,
+  pubKeyfromPrivKey,
+} from '@stacks/transactions';
+
+// for mainnet, use `StacksMainnet()`
+const network = new StacksTestnet();
+const fullyQualifiedName = 'name.id';
+const salt = 'salt';
+const stxToBurn = 200n;
+const privateKey = 'cb3df38053d132895220b9ce471f6b676db5b9bf0b4adefb55f2118ece2478df01';
```
Comment (diwakergupta) [inline packages/bns/README.md]: I'm guessing this is fine for test code but we don't want to encourage people to hard-code in their private keys in production.

Is it worth putting in a caveat or providing some best-practices @reedrosenbluth ?

**Review by diwakergupta — APPROVED** [review]
LGTM but would like Reed's feedback on the private key comment.

File: packages/bns/README.md (hunk 1 above)
Comment (reedrosenbluth) [reply packages/bns/README.md]: yes, definitely!

**Review by reedrosenbluth — APPROVED** [review]
LGTM!

File: packages/bns/README.md (hunk 1 above)
Comment (ahsan-javaid) [reply packages/bns/README.md]: Added caveats for `privateKey `. Let me know if it's fine or need any further improvements.

**Comment by ahsan-javaid** [comment]
> LGTM!

Added caveats for `privateKey `. Let me know if it's fine or need any further improvements.

**Comment by agraebe** [comment]
@reedrosenbluth can this be merged?

**Pull request merged**